autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -504,13 +489,16 @@ def impute_nan_in_categories(self, X: pd.DataFrame) -> pd.DataFrame:
         # TypeError: '<' not supported between instances of 'int' and 'str'
         # in the encoding
         for column in self.enc_columns:
-            if X[column].isna().any():
+            # no missing values for categorical column
+            if not X[column].isna().any():
+                continue
+            else:
```

Comment:
Ah thanks for addressing my comment:)

If you need to use `else` later, you can go back to the original version.
Just in case, what I was thinking was to remove the `else` completely, because we do not nest for this `else` after the line continuation.

Reply:
Yes I understood it, I have a bit of time without checking code coverage tools, but if I remember correctly when you only have an if and not the else, it counts your code as not fully covered, so for the sake of that and a nice metric value in the future, I was thinking of adding the else, as it is pretty redundant. If it is not the case now, we can remove it.

Reply:
Sure, I did not know [the bug](https://github.com/nedbat/coveragepy/issues/198).
Then it is better like this based on the report?:
```
if skip:
    _dummy = True  # Heuristics to avoid a non-coverage issue
    continue

func1()
func2()
...
```

Reply:
They say they addressed it so it should be fine for the future.